src/interpreter/kernel/instruction/unop.js

```diff
@@ -31,6 +38,9 @@ function unop(
 
     case "eqz":
       return createValue(value.eqz());
+
+    case "reinterpret/f32":
+      return i32.createValue(value.reinterpret());
```

Comment:
this breaks the current pattern, which expects all operations to return the same type as their operand(s). Also, this shouldn't be hard-coded as `i32`

Reply:
Yeah interesting, it returns an opaque `StackLocal` tho. I think there's nothing conceptually wrong, but just confusing.

Reply:
Yeah, I think it bends the current patten a little, but not too much.